internal/build/custom_builder.go

```diff
@@ -0,0 +1,86 @@
+package build
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/docker/distribution/reference"
+	"github.com/opencontainers/go-digest"
+
+	"github.com/windmilleng/tilt/internal/docker"
+	"github.com/windmilleng/tilt/internal/logger"
+)
+
+type CustomBuilder interface {
+	Build(ctx context.Context, ref reference.Named, command string) (reference.NamedTagged, error)
+}
+
+type ExecCustomBuilder struct {
+	dCli  docker.Client
+	env   []string
+	clock Clock
+}
+
+func NewExecCustomBuilder(dCli docker.Client, env docker.Env, clock Clock) *ExecCustomBuilder {
+	return &ExecCustomBuilder{
+		dCli:  dCli,
+		env:   env.BuildInjections,
+		clock: clock,
+	}
+}
+
+func (b *ExecCustomBuilder) Build(ctx context.Context, ref reference.Named, command string) (reference.NamedTagged, error) {
+	tmpTag := fmt.Sprintf("tilt-build-%d", b.clock.Now().Unix())
+	result, err := reference.WithTag(ref, tmpTag)
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := exec.CommandContext(ctx, "sh", "-c", command)
+
+	l := logger.Get(ctx)
+	l.Infof("Custom Build: Injecting Environment Variables")
```

Comment:
this looks like potentially leftover printf debugging

Reply:
If we don't have this, it's hard to understand what's happening when you're trying to set up your custom build.

E.g., you run a command, but it doesn't leave a docker image on your docker daemon because we set DOCKER_HOST, and you're like WTF.